### ISSUE TITLE
API change between versions 3.1.0 and 3.2.1?

### ISSUE THREAD

**Issue by ethan-gerardot** [TheLevelUp/ZXingObjC#334]
Everything was working with 3.1.0, but I upgraded to 3.2.1 and now the delegate method `captureResult:result:` isn't getting called.
Nothing stands out when I look at the BarcodeScanner example.
My basic setup is like this (and it worked great with 3.1.0):

    - (void)viewDidLoad {
        [super viewDidLoad];
        
        if (_capture == nil) {
            self.capture = [[ZXCapture alloc] init];
        }
        self.capture.camera = _capture.back;
        self.capture.focusMode = AVCaptureFocusModeContinuousAutoFocus;
        self.capture.rotation = 90.0f;
        
        self.capture.layer.frame = self.view.bounds;
        [self.view.layer addSublayer:_capture.layer];
        
        [self.view bringSubviewToFront:_scanRectView];
        
        self.capture.delegate = self;
    }
     
    - (void)viewDidAppear:(BOOL)animated {
        [super viewDidAppear:animated];
        
        // Start running the ZXCapture.
        if (!_capture.running) {
            [_capture start];
        }
        
        self.capture.layer.frame = self.view.bounds;
        
        // Bring the camera layer to the front to view the input coming through the camera.
        [self.capture.layer removeFromSuperlayer];
        [self.view.layer insertSublayer:_capture.layer atIndex:([[self.view.layer sublayers] count] - 1.0)];
        
        CGAffineTransform captureSizeTransform = CGAffineTransformMakeScale(
                            self.scanRectView.frame.size.width / self.view.frame.size.width,
                            self.scanRectView.frame.size.height / self.view.frame.size.height);
    
        self.capture.scanRect = CGRectApplyAffineTransform(_scanRectView.frame, captureSizeTransform);
    }

**Comment by ethan-gerardot** [comment]
I updated my code to more closely match the BarcodeScanner example, and now it's working. I still think it would be helpful if someone knows and would like to point out what is wrong with my code above for version 3.2.1 (since it worked in 3.1.0).

**Comment by benjohnde** [comment]
What did you change? :)

**Comment by benjohnde** [comment]
Comparison of both versions: https://github.com/TheLevelUp/ZXingObjC/compare/3.2.0...3.2.1

Would be interesting to know which barcode you are trying to scan and what are the other environment variables, for instance which device you use, which operating system runs on it.

I guess it have something to do with the different session preset, but as said, it would be very helpful to know which barcode you are trying to scan and what are the other environment variables such as operating system.

**Comment by marcuswu0814** [comment]
@ethan-gerardot Could you share what did you change? I got same problem, and remove `scanRect` it work.

**Comment by benjohnde** [comment]
Any progress on here?

**Comment by ethan-gerardot** [comment]
Sorry I never responded on this...it dropped off my radar because when I updated my code to match the example, it fixed the issue.

I have since delved a bit deeper into making my scan rect more accurate and making the setup code simpler in my view controller. I think the issue was that the newer API required some transforms / rotation stuff on the capture. I wouldn't consider that to be a bug necessarily, but maybe there's some other issue I'm not aware of. Anyways, the new setup code makes sense to me and I no longer have any issues/confusion with the changes that went into 3.2.1.
